packages/manager/src/features/linodes/LinodesDetail/LinodeNetworking/IPSharing.tsx

```diff
@@ -154,36 +193,110 @@ const IPSharingPanel: React.FC<CombinedProps> = (props) => {
   };
 
   const remainingChoices = (selectedIP: string): string[] => {
-    return ipChoices.filter((ip: string) => {
+    return Object.keys(ipChoices).filter((ip: string) => {
       const hasBeenSelected = ipsToShare.includes(ip);
       return ip === selectedIP || !hasBeenSelected;
     });
   };
 
   const onSubmit = () => {
-    const finalIPs = uniq(ipsToShare.filter(Boolean));
-
+    const groupedUnsharedRanges = {};
+    const finalIPs: string[] = uniq(
+      ipsToShare.reduce((previousValue, currentValue) => {
+        if (currentValue === undefined || currentValue === null) {
+          return previousValue;
+        }
+        const strippedIP: string = currentValue.includes('/')
+          ? currentValue.substr(0, currentValue.indexOf('/'))
+          : currentValue;
```

Comment:
```suggestion
        const strippedIP: string = currentValue.split('/')[0];
```

I think this is equivalent.

Reply:
nice